src/backends/plonky2/mock/signedpod.rs

```diff
@@ -138,17 +141,17 @@ pub mod tests {
         let pod = pod.sign(&mut signer).unwrap();
         let pod = pod.pod.into_any().downcast::<MockSignedPod>().unwrap();
 
-        assert!(pod.verify());
+        pod.verify()?;
         println!("id: {}", pod.id());
         println!("kvs: {:?}", pod.kvs());
 
         let mut bad_pod = pod.clone();
         bad_pod.signature = "".into();
-        assert!(!bad_pod.verify());
+        bad_pod.verify().is_err();
```

Comment:
```suggestion
        assert!(bad_pod.verify().is_err());
```
And the same applies to the following cases

Reply:
oh right! my bad; thanks for catching it!, updated it, also used the opportunity to rebase this PR branch to latest `main` branch version.